Framework/HistogramData/src/HistogramItem.cpp

```diff
@@ -0,0 +1,106 @@
+#include "MantidHistogramData/HistogramItem.h"
+#include "MantidHistogramData/Histogram.h"
+#include <algorithm>
+#include <utility>
+
+using Mantid::HistogramData::HistogramItem;
+using Mantid::HistogramData::Histogram;
+using Mantid::HistogramData::BinEdges;
+using Mantid::HistogramData::Points;
+
+HistogramItem::HistogramItem(const Histogram &histogram, const size_t index)
+    : m_histogram(histogram), m_index(index) {}
+
+double HistogramItem::counts() const {
+  const auto &yMode = histogramRef().yMode();
+  if (yMode == Histogram::YMode::Counts) {
+    return histogramRef().counts()[m_index];
+  } else {
+    const Frequencies frequency{histogramRef().frequencies()[m_index]};
+    return Counts(frequency, binEdges())[0];
+  }
+}
+
+double HistogramItem::countVariance() const {
+  return histogramRef().countVariances()[m_index];
+}
+
+double HistogramItem::countStandardDeviation() const {
+  return histogramRef().countStandardDeviations()[m_index];
+}
+
+double HistogramItem::frequency() const {
+  const auto &yMode = histogramRef().yMode();
+  if (yMode == Histogram::YMode::Frequencies) {
+    return histogramRef().frequencies()[m_index];
+  } else {
+    const Counts counts{histogramRef().counts()[m_index]};
+    return Frequencies(counts, binEdges())[0];
```

Comment:
Similar to what I said above: For performance reasons we cannot rely on all the math done by classes like `Frequencies`, you need to implement this yourself, based on the underlying data.

Reply:
Again, I thought it might be better to let the Histogram handle the computations itself, but if there's a likely to be a large performance hit then I'll do it manually.